packages/core/src/decorators/common/remove-x-internal.ts

```diff
@@ -6,39 +6,50 @@ import type { UserContext } from '../../walk';
 
 const DEFAULT_INTERNAL_PROPERTY_NAME = 'x-internal';
 
-export const RemoveXInternal: Oas3Decorator | Oas2Decorator = ({ internalFlagProperty }) => {
-  const hiddenTag: string = internalFlagProperty || DEFAULT_INTERNAL_PROPERTY_NAME;
-
-  function removeInternal(node: any, ctx: UserContext) {
+export const RemoveXInternal: Oas3Decorator | Oas2Decorator = ({
+  internalFlagProperty = DEFAULT_INTERNAL_PROPERTY_NAME,
+}) => {
+  function removeInternal(
+    node: unknown,
+    ctx: UserContext,
+    originalMapping: Record<string, string>
+  ) {
     const { parent, key } = ctx;
     let didDelete = false;
     if (Array.isArray(node)) {
       for (let i = 0; i < node.length; i++) {
         if (isRef(node[i])) {
           const resolved = ctx.resolve(node[i]);
-          if (resolved.node?.[hiddenTag]) {
+          if (resolved.node?.[internalFlagProperty]) {
+            // First, remove the reference in the discriminator mapping, if it exists:
+            if (parent.discriminator?.mapping) {
```

Comment:
add extra check for typeof object

Reply:
Added [here](https://github.com/Redocly/redocly-cli/pull/1694/commits/03f55f8f73bb4509a06a0f648460fa61572fd43e).